astropy/io/fits/fitstime.py

```diff
@@ -0,0 +1,397 @@
+                AstropyUserWarning)
+            column_info['TCTYP'] = global_info['TIMESYS']
+            column_info['FORMAT'] = global_info.get('FORMAT', None)
+
+        # Non-linear coordinate types have "4-3" form and are not time coordinates
+        if TCTYP_RE_TYPE.match(scale[:5]) and TCTYP_RE_ALGO.match(scale[5:]):
+            return False
+
+        # Coordinate type is either an unrecognized local time scale
+        # or a linear coordinate type
+        return False
+
+    if global_info['TIMESYS'] is None:
+        return False
+
+    if (unit is not None and unit in FITS_TIME_UNIT) or pos is not None:
+        column_info['TCTYP'] = global_info['TIMESYS']
+        return True
+
+    return False
+
+
+def _convert_global_time(table, global_info):
+    """
+    Convert the table metadata for time informational keywords
+    to astropy Time.
+
+    Parameters
+    ----------
+    table : `~astropy.table.Table`
+        The table whose time metadata is to be converted.
+    global_info : dict
+        Global time reference frame information.
+    """
+
+    # Read in Global Informational keywords as Time
+    for key, value in global_info.items():
+        # FITS uses a subset of ISO-8601 for DATE-xxx
+        if key.startswith('DATE'):
+            if key not in table.meta:
+                scale = 'utc' if key == 'DATE' else global_info['TIMESYS']
+                try:
+                    precision = len(value.split('.')[-1]) if '.' in value else 0
+                    value = Time(value, format='fits', scale=scale,
+                                 precision=precision)
+                except ValueError:
+                    pass
+                table.meta[key] = value
+
+        # MJD-xxx in MJD according to TIMESYS
+        elif key.startswith('MJD-'):
+            if key not in table.meta:
+                try:
+                    value = Time(value, format='mjd',
+                                 scale=global_info['TIMESYS'])
+                except ValueError:
+                    pass
+                table.meta[key] = value
+
+
+def _convert_time_columns(col, column_info):
+    """
+    Convert time columns to astropy Time columns.
+
+    Parameters
+    ----------
+    col : `~astropy.table.Column`
+        The time coordinate column to be converted to Time.
+    column_info : dict
+        Column-specific time reference frame override information.
+    """
+
+    # The code might fail while attempting to read FITS files not written by astropy.
+
+    # Read in time coordinate column as Time
+    if col.shape[-1] == 2 and col.ndim > 1:
+        col = Time(col[..., 0], col[..., 1], format='jd',
+                   scale=column_info['TCTYP'], location=column_info['TRPOS'])
+    else:
+        warnings.warn(
+            'Time column {} is not in the astropy required (jd1, jd2) format. '
+            'Hence, it will not be read as astropy Time'.format(col.info.name),
+            AstropyUserWarning)
+        return
```

Comment:
You have to return `col` here (and outside the `else` clause), otherwise every column will become `None`!!!